drivers/console/bm_serial.c

```diff
@@ -1,27 +1,27 @@
 /** @file
- * @brief Pipe UART driver
+ * @brief Bristlemouth Serial driver
  *
- * A pipe UART driver allowing application to handle all aspects of received
- * protocol data.
+ * A Bristlemouth Serial driver to send and receive Manchester Encoded packets
+ * using UART DMA
  */
 
 /*
  * Copyright (c) 2015 Intel Corporation
+ * Copyright (c) 2022 Sofar Ocean Technologies
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include <logging/log.h>
 LOG_MODULE_REGISTER(bm_serial, CONFIG_UART_CONSOLE_LOG_LEVEL);
 
-#include <kernel.h>
 #include <drivers/uart.h>
 #include <sys/printk.h>
 #include <drivers/console/bm_serial.h>
 #include <sys/crc.h>
 #include <drivers/gpio.h>
 
-static const struct device* serial_dev[3] = {0};
+static bm_ctx_t dev_ctx[CONFIG_BM_MAX_SERIAL_DEV_COUNT];
```

Comment:
Pretty much all of the of the globals in this file, as far as I can tell, are referenced in ISRs or are used in multiple threads, and should be marked as volatile.

Similarly, all globals defined here are only used in this .c file, so should be marked static to avoid potential multiple definition issues in the future.